File: main_module/src/main/java/com/phone/main_module/login/presenter/ILoginPresenter.java
package com.phone.main_module.login.presenter;

import com.trello.rxlifecycle3.components.support.RxAppCompatActivity;

import java.util.Map;

/**
 * author    : Urasaki
 * e-mail    : 1164688204@qq.com
 * date      : 2019/3/10 16:58
 * introduce :
 */

public interface ILoginPresenter {

    void getAuthCode(RxAppCompatActivity rxAppCompatActivity, Map<String, String> bodyParams);

    void loginWithAuthCode(RxAppCompatActivity rxAppCompatActivity, Map<String, String> bodyParams);

    void login(RxAppCompatActivity rxAppCompatActivity, Map<String, String> bodyParams);

    void register(RxAppCompatActivity rxAppCompatActivity, Map<String, String> bodyParams);

}

File: library_login/src/main/java/com/phone/library_login/presenter/ILoginPresenter.kt
package com.phone.library_login.presenter

import com.trello.rxlifecycle3.components.support.RxAppCompatActivity

/**
 * author    : Urasaki
 * e-mail    : 1164688204@qq.com
 * date      : 2019/3/10 16:58
 * introduce :
 */

interface ILoginPresenter {

    fun getAuthCode(rxAppCompatActivity: RxAppCompatActivity, bodyParams: Map<String, String>)

    fun loginWithAuthCode(
        rxAppCompatActivity: RxAppCompatActivity,
        bodyParams: Map<String, String>
    )

    fun login(rxAppCompatActivity: RxAppCompatActivity, bodyParams: Map<String, String>)

    fun register(rxAppCompatActivity: RxAppCompatActivity, bodyParams: Map<String, String>)

}